### PR TITLE
Fix GCC format-signedness warnings

### PR DESCRIPTION
**Pull request by kennethsoerensen**

### CODE REVIEW

File: library/error.c [hunk 1]
```diff
@@ -404,7 +404,7 @@ void mbedtls_strerror( int ret, char *buf, size_t buflen )
         // END generated code
 
         if( strlen( buf ) == 0 )
-            mbedtls_snprintf( buf, buflen, "UNKNOWN ERROR CODE (%04X)", use_ret );
+            mbedtls_snprintf( buf, buflen, "UNKNOWN ERROR CODE (%04X)", (unsigned int)use_ret );
```
Comment (gilles-peskine-arm) [inline library/error.c]: `library/error.c` is automatically generated. Please edit `scripts/data_files/error.fmt` instead, then run `scripts/generate_errors.pl`.

File: library/ecp.c [hunk 2]
```diff
@@ -3128,7 +3128,7 @@ int mbedtls_ecp_self_test( int verbose )
 cleanup:
 
     if( ret < 0 && verbose != 0 )
-        mbedtls_printf( "Unexpected error, return code = %08X\n", ret );
+        mbedtls_printf( "Unexpected error, return code = %08X\n", (unsigned int)ret );
```
Comment (gilles-peskine-arm) [inline library/ecp.c]: Our house style is to put a space after the cast.

**Comment by kennethsoerensen** [comment]
I am sorry but I can't figure if I need to do more on this pull request?

I can see that some checks have failed but I can't figure if this is expected?

**Comment by gilles-peskine-arm** [comment]
Thanks for updating. The CI has now passed except for some Mbed OS job which are a preexisting issue, so the PR is good on this front. We now need to review it.

**Comment by gilles-peskine-arm** [comment]
For legal reasons, we cannot merge this pull request unless you indicate that we are allowed to redistribute your contribution by including a Signed-off-by: line in each commit message. For more information, see https://github.com/ARMmbed/mbed-crypto/issues/371 and [`CONTRIBUTING.md`](https://github.com/ARMmbed/mbed-crypto/blob/development/CONTRIBUTING.md).

**Comment by gilles-peskine-arm** [comment]
[We've now merged Mbed Crypto back into Mbed TLS](https://github.com/ARMmbed/mbed-crypto/issues/374). Since this pull request changes the build options, we won't be able to merge it directly into Mbed TLS: that would be likely to cause build failures due to signedness issues in x509 and tls files. Therefore I'm closing this pull request here in Mbed Crypto. Please raise a new pull request in Mbed TLS that also fixes issues in non-crypto files.

**Comment by kennethsoerensen** [comment]
Please see https://github.com/ARMmbed/mbedtls/pull/3153

**Pull request closed without merging**